rskj-core/src/main/java/co/rsk/peg/PegUtilsLegacy.java

```diff
@@ -73,16 +73,16 @@ protected static boolean isPegOutTx(BtcTransaction btcTx, ActivationConfig.ForBl
         int inputsSize = btcTx.getInputs().size();
         for (int i = 0; i < inputsSize; i++) {
             TransactionInput txInput = btcTx.getInput(i);
-            Optional<Script> redeemScriptOptional = BitcoinUtils.extractRedeemScriptFromInput(btcTx.getInput(i));
+            Optional<Script> redeemScriptOptional = BitcoinUtils.extractRedeemScriptFromInput(txInput);
             if (!redeemScriptOptional.isPresent()) {
                 continue;
             }
 
             List<ScriptChunk> redeemScriptChunks = redeemScriptOptional.get().getChunks();
             if (activations.isActive(ConsensusRule.RSKIP201)) {
                 // Extract standard redeem script chunks since the registered utxo could be from a fast bridge or erp federation
-                RedeemScriptParser redeemScriptParser = RedeemScriptParserFactory.get(txInput.getScriptSig().getChunks());
                 try {
+                    RedeemScriptParser redeemScriptParser = RedeemScriptParserFactory.get(redeemScriptChunks);
                     redeemScriptChunks = redeemScriptParser.extractStandardRedeemScriptChunks();
```

Comment:
Just this line

``` RedeemScriptParser redeemScriptParser = RedeemScriptParserFactory.get(redeemScriptChunks); ```

Reply:
Update:

To move the second line out of the try, I will also need to move the `RedeemScriptParser redeemScriptParser` declaration out of the try. it will look like this:

```
...
      List<ScriptChunk> redeemScriptChunks = redeemScriptOptional.get().getChunks();
            if (activations.isActive(ConsensusRule.RSKIP201)) {
                // Extract standard redeem script chunks since the registered utxo could be from a fast bridge or erp federation
                RedeemScriptParser redeemScriptParser;
                try {
                    redeemScriptParser = RedeemScriptParserFactory.get(redeemScriptChunks);
                } catch (ScriptException e) {
                    logger.debug("[isPegOutTx] There is no redeem script", e);
                    // There is no redeem script
                    continue;
                }
                redeemScriptChunks = redeemScriptParser.extractStandardRedeemScriptChunks();
            }

            Script redeemScript = new ScriptBuilder().addChunks(redeemScriptChunks).build();
...

```

That way looks more complex to me. I prefer to keep it as it is now.